state-synchronizer/src/logging.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{chunk_request::GetChunkRequest, chunk_response::GetChunkResponse};
+use anyhow::Error;
+use libra_config::config::PeerNetworkId;
+use libra_logger::Schema;
+use libra_types::{
+    contract_event::ContractEvent, ledger_info::LedgerInfoWithSignatures, waypoint::Waypoint,
+};
+use serde::Serialize;
+#[derive(Clone, Schema)]
+pub struct LogSchema<'a> {
+    name: LogEntry,
+    event: Option<LogEvent>,
+    #[schema(debug)]
+    error: Option<&'a Error>,
+    #[schema(display)]
+    peer: Option<PeerNetworkId>,
```

Comment:
You should be able to do `Option<&'a PeerNetworkId>` then you don't have to have the custom clone function you have below.